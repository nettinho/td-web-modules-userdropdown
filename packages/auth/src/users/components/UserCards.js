import React from "react";
import { Icon, Card } from "semantic-ui-react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import { FormattedMessage } from "react-intl";
import routes from "../routes";
import UserCard from "./UserCard";

export const UserCards = ({ users }) => (
  <Card.Group>
    <Card href={routes.USER_CREATE}>
      <Card.Content textAlign="center">
        <Card.Header>
          <FormattedMessage id="users.actions.create" />
        </Card.Header>
        <Card.Description>
          <Icon floated="center" name="add user" size="massive" />
        </Card.Description>
      </Card.Content>
    </Card>
    {users.map((user, i) => <UserCard key={i} user={user} />)}
  </Card.Group>
);

UserCards.propTypes = {
  users: PropTypes.array
};

const mapStateToProps = ({ users }) => ({ users });

export default connect(mapStateToProps)(UserCards);
