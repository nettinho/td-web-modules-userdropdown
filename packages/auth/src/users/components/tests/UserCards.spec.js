import React from "react";
import { shallow } from "enzyme";
import { UserCards } from "../UserCards";

describe("<UserCards />", () => {
  const deleteUser = jest.fn();
  const users = [{ id: 1 }, { id: 2 }];
  const props = { users, deleteUser };

  it("matches the latest snapshot", () => {
    const wrapper = shallow(<UserCards {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
