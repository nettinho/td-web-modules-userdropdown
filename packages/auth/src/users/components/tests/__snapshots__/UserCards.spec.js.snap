// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserCards /> matches the latest snapshot 1`] = `
<CardGroup>
  <Card
    href="/users/new"
  >
    <CardContent
      textAlign="center"
    >
      <CardHeader>
        <FormattedMessage
          id="users.actions.create"
          values={Object {}}
        />
      </CardHeader>
      <CardDescription>
        <Icon
          as="i"
          floated="center"
          name="add user"
          size="massive"
        />
      </CardDescription>
    </CardContent>
  </Card>
  <Connect(UserCard)
    key="0"
    user={
      Object {
        "id": 1,
      }
    }
  />
  <Connect(UserCard)
    key="1"
    user={
      Object {
        "id": 2,
      }
    }
  />
</CardGroup>
`;
